### DISCUSSION TITLE
dyld[59428]: Library not loaded

### DISCUSSION THREAD

**Discussion by tekumara** [python-poetry/poetry#8298] [Q&A] [answered]
After a brew update:
```
❯ poetry
dyld[59428]: Library not loaded: /opt/homebrew/Cellar/python@3.9/3.9.17/Frameworks/Python.framework/Versions/3.9/Python
  Referenced from: <F922A9DD-D860-3DDE-9FC0-17C5472BA5D2> /Users/tekumara/Library/Application Support/pypoetry/venv/bin/python
  Reason: tried: '/opt/homebrew/Cellar/python@3.9/3.9.17/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/python@3.9/3.9.17/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/opt/homebrew/Cellar/python@3.9/3.9.17/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file, not in dyld cache)
zsh: abort      poetry
```

**Comment by tekumara** [comment, marked as answer]
Resolved by uninstalling and reinstalling:
```
curl -sSL https://install.python-poetry.org | python3 - --uninstall
curl -sSL https://install.python-poetry.org | python3 -
```

**Reply by thatbeautifuldream** [reply]
worked for me, thanks!

**Reply by elinarezv** [reply]
Work for me too!!

**Comment by neilpvirtualitics** [comment]
sadly; the re-install of poetry results in this error for me:

```
dyld[3691]: Library not loaded: @loader_path/../../../../Python.framework/Versions/3.8/Python
  Referenced from: <17C7651B-E7AD-3F57-96B6-A18344D8BF29> /Users/neil.p/Library/Application Support/pypoetry/venv/bin/python3
  Reason: tried: '/Users/neil.p/Library/Application Support/pypoetry/venv/bin/../../../../Python.framework/Versions/3.8/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.8/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.8/Python' (no such file, not in dyld cache)

Traceback:

  File "./install-poetry.py", line 923, in main
    return installer.run()
  File "./install-poetry.py", line 562, in run
    raise PoetryInstallationError(
```

OS ver: Ventura 13.6
system python version 3.9.6
brew python versions: python@3.11, python@3.10, python@3.8, python@3.9 (these can't be uninstalled without deleting other dependencies, though I normally have 3.8 (3.8.18) linked.  (via .bash_profile; I link /usr/local/Cellar/python@3.8/3.8.18/Frameworks/Python.framework/Versions/3.8/bin/python3.8 to /usr/local/opt/python, /usr/local/bin/python, /usr/local/bin/python3, and /usr/local/bin/python3.8, and I leave /usr/bin/python3 alone, for the system-installed 3.9.6).

I have an empty ~/Library/Application Support/pypoetry directory, that gets re-created when I try to re-install, but it's left empty.

There is no /Library/Frameworks/Python.framework directory on my system. There is also no /System/Library/Frameworks/Python.framework directory on my system.  I don't know if the 13.6 upgrade I did deleted these, or what.  It's very unlikely I removed those, but that's probably the source of my 'no such file' errors. I can still selectively run python3 to get my 3.8.18 version, and run /usr/bin/python3 to get 3.9.6.  And I don't even know how that's possible with those directories gone.

**Reply by neilpvirtualitics** [reply]
How I fixed it (roughly):
updated to mac os x 14 (Sonoma).
uninstalled all of my pip packages.
uninstalled all of my brew dependencies for python@3.11, @3.10, (brew's) @3.9, and @3.8.
uninstalled all of those python versions (brew).
important! -> app store, updated XCode.
brew install python@3.8
poetry installs fine now.

Apparently, if XCode doesn't match your OS version that dyld cache is messed up.  I don't think I updated xcode over my last 3 OS updates (at least).  Be aware that downloading the xcode update from the app store is . . . rather time-consuming.

**Comment by tekumara** [comment]
FYI - the other possibility is when using homebrew note that the [homebrew python binary links to a relative path](https://github.com/python-poetry/install.python-poetry.org/issues/24#issuecomment-1226504499). Because the poetry install script creates venvs by copying the python binary, the python binary can't load in its new location and fails with the error above.
